applications/IgaApplication/custom_conditions/sbm_dirichlet_laplacian_condition.cpp

```diff
@@ -0,0 +1,378 @@
+            for (IndexType n = 1; n <= mBasisFunctionsOrder; n++) {
+                Matrix& r_shape_function_derivatives = shape_function_derivatives[n-1];
+                
+                int countDerivativeId = 0;
+                // Loop over blocks of derivatives in x
+                for (IndexType k_x = n; k_x >= 0; k_x--) {
+                    // Loop over the possible derivatives in y
+                    for (IndexType k_y = n - k_x; k_y >= 0; k_y--) {
+                        
+                        // derivatives in z
+                        IndexType k_z = n - k_x - k_y;
+                        double derivative = r_shape_function_derivatives(i,countDerivativeId); 
+
+                        H_taylor_term += ComputeTaylorTerm3D(derivative, mDistanceVector[0], k_x, mDistanceVector[1], k_y, mDistanceVector[2], k_z);
+                        countDerivativeId++;
+                    }
+                }
+            }
+        }
+        H_sum_vec(i) = H_taylor_term + N(0,i);
+    }
+}
+
+unsigned long long SbmDirichletLaplacianCondition::factorial(IndexType n) 
+{
+    if (n == 0) return 1;
+    unsigned long long result = 1;
+    for (IndexType i = 2; i <= n; ++i) result *= i;
+    return result;
+}
+
+// Function to compute a single term in the Taylor expansion
+double SbmDirichletLaplacianCondition::ComputeTaylorTerm(double derivative, double dx, IndexType n_k, double dy, IndexType k)
+{
+    return derivative * std::pow(dx, n_k) * std::pow(dy, k) / (factorial(k) * factorial(n_k));    
+}
+
+double SbmDirichletLaplacianCondition::ComputeTaylorTerm3D(double derivative, double dx, IndexType k_x, double dy, IndexType k_y, double dz, IndexType k_z)
+{   
+    return derivative * std::pow(dx, k_x) * std::pow(dy, k_y) * std::pow(dz, k_z) / (factorial(k_x) * factorial(k_y) * factorial(k_z));    
+}
+
+
+int SbmDirichletLaplacianCondition::Check(const ProcessInfo& rCurrentProcessInfo) const
+{
+    KRATOS_ERROR_IF_NOT(GetProperties().Has(PENALTY_FACTOR))
+        << "No penalty factor (PENALTY_FACTOR) defined in property of SbmDirichletLaplacianCondition" << std::endl;
+    return 0;
+}
+
+void SbmDirichletLaplacianCondition::EquationIdVector(
+    EquationIdVectorType& rResult,
+    const ProcessInfo& rCurrentProcessInfo
+) const
+{
+    ConvectionDiffusionSettings::Pointer p_settings = rCurrentProcessInfo[CONVECTION_DIFFUSION_SETTINGS];
+    const auto& r_unknown_var = p_settings->GetUnknownVariable();
+
+    const auto& r_geometry = GetGeometry();
+    const SizeType number_of_nodes = r_geometry.size();
+
+    if (rResult.size() !=  number_of_nodes)
+        rResult.resize(number_of_nodes, false);
+
+    for (IndexType i = 0; i < number_of_nodes; ++i) {
+        const auto& r_node = r_geometry[i];
+        rResult[i] = r_node.GetDof(r_unknown_var).EquationId();
+    }
+}
+
+void SbmDirichletLaplacianCondition::GetDofList(
+    DofsVectorType& rElementalDofList,
+    const ProcessInfo& rCurrentProcessInfo
+) const

```

Comment:
```suggestion
    const ProcessInfo& rCurrentProcessInfo) const
```
(just to be consistent with the rest of the file)